pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go

```diff
@@ -200,7 +204,12 @@ func (c *defaultControllerInstallationControl) reconcile(controllerInstallation
 		return err
 	}
 
-	release, err := chartRenderer.RenderArchive(helmDeployment.Chart, controllerRegistration.Name, common.GardenNamespace, helmDeployment.Values)
+	namespace := getNamespaceForControllerInstallation(controllerInstallation)
+	if err := k8sSeedClient.Client().Create(context.TODO(), namespace); err != nil && !apierrors.IsAlreadyExists(err) {
+		return err
+	}
+
+	release, err := chartRenderer.RenderArchive(helmDeployment.Chart, controllerRegistration.Name, namespace.Name, helmDeployment.Values)
```

Comment:
Just happened to pass by here - shouldn't it be non-mandatory for a controller installation to be a helm deployment? As far as I can tell from here, it currently isn't...

Reply:
No, it's completely optional, see https://github.com/rfranzke/gardener/blob/feature/incorporate-coreos-extension/pkg/controllermanager/controller/controllerinstallation/controllerinstallation_control.go#L138-L140.